### DISCUSSION TITLE
[Feature Request] Support for Multiple Cursos (apply Carry Forward Line-by-Line)

### DISCUSSION THREAD

**Discussion by FelipeRearden** [jglev/obsidian-carry-forward#11] [Ideas]
Hello @jglev !!!!!!!!!!!!

My FR is simple to explain. Would be amazing if we could use Carry-Forward when we have a multiple cursors selected 

Let me explain mora about this with an example ....

## My Workflow

I have web articles as .md notes inside Obsidian and I use the highlight command `==text==` to highlight parts of the text 

The plugin Advanced Cursors (https://github.com/SkepticMystic/advanced-cursors) have a feature that let the users create a regex query to select ALL entries from the query:
- This way I have all my highlights SELECTED as multiple cursors 
- If something is selected, I can use a COPY/CUT command to transfer this content to any note that I want

Copying ...... -> That's where our beloved Carry-Forward can help me :)

After selecting all highlights content with an Advanced Cursors query I want to use `Carry-Forward:Copy Selection with Each Line Linked to its copied source (default link text)` to copy the selected content AND create block ref to each line that has content selected . *This is my FR !!!!!!*

In my case, after this, I will use our super Apply Patterns to insert line breaks between each highlight using the new command `Whole Clipboard (link-by-line)`
- `(.+)` ---> `\n{1}`

## Carry-Forward Behavior today vs. My FR

This is my Note with highlights 

![0](https://user-images.githubusercontent.com/85180696/150780018-55257222-5729-4e01-87e2-38bb6a273818.png)

This is my Advanced Cursors Query to select the highlights `/==.*?==/m`

![1](https://user-images.githubusercontent.com/85180696/150780040-32db759b-f155-410e-a2e8-b1a88eb60aa1.png)

Using the Command Carry-Forward: `Carry-Forward:Copy Selection with Each Line Linked to its copied source (default link text)` 

![2](https://user-images.githubusercontent.com/85180696/150780057-bf563db3-2fe8-4c8e-ba30-727cc76f3a24.png)

As you can see:,
- after hitting the command, only the first cursor is selected ;
- and it creates the block refs from from the first selected item :(

This is the result that I'm looking for :pray: !!!!!!!

![3](https://user-images.githubusercontent.com/85180696/150780080-3cc04443-e4b5-495b-93d3-e6124fae1392.png)

one more info: the standard Copy (CRTL/CMD+C) is able to copy content from multiple cursors :)

![4](https://user-images.githubusercontent.com/85180696/150781670-f4febfac-3638-4a15-ba67-52dee2621afc.png)
 
---

In my mind, what I'm asking is something related to the commands `whole lines` and `clipboard (line-by-line)` in a way that Carry-Forward command is applied to each selection 

I don't know if what I'm asking is feasible :pray:

Thanks again for reading !!!!! I hope you like this idea as much as I do 

I wish you a fantastic Monday!!!!!

**Comment by jglev** [comment]
@FelipeRearden, as always, thank you for writing so thoughtfully and thoroughly!!

I've tried addressing this in the upcoming [`v1.4.0`](https://github.com/jglev/obsidian-carry-forward/releases/tag/1.4.0-beta0) release, on your suggestion : ) I'd be grateful for feedback!

**Reply by FelipeRearden** [reply]
> @FelipeRearden, as always, thank you for writing so thoughtfully and thoroughly!!

Thank you very much for your feedback @jglev !!!!

> I've tried addressing this in the upcoming [`v1.4.0`](https://github.com/jglev/obsidian-carry-forward/releases/tag/1.4.0-beta0) release, on your suggestion : ) I'd be grateful for feedback!

Wooowwww !!!!!!! Sure !!!! It's my pleasure doing the test for this beta version. It's bed time here but I'm going to start my tests first thing in the morning and I get back to you :)

I'm so happy !!!!! Thank you very much for making me dream come true :) 

We talk tomorrow :)
